src/main/java/net/minecraft/launchwrapper/LaunchClassLoader.java

```diff
@@ -384,4 +393,12 @@ private static void closeSilently(Closeable closeable) {
     public void clearNegativeEntries(Set<String> entriesToClear) {
         negativeResourceCache.removeAll(entriesToClear);
     }
+
+    /**
+     * @author soniex2
+     */
+    private static final class LaunchData {
+        private List<IClassTransformer> transformers = new ArrayList<IClassTransformer>(2);
+        private IClassNameTransformer renameTransformer;
```

Comment:
Remove `private` to prevent bridge method generations.

Reply:
why would it generate bridge methods?

Reply:
It wouldn't generate bridge methods, although it would generate synthetic methods [due to how inner classes work](http://www.benf.org/other/cfr/inner-class-fake-friends.html).  Whether or not this really maters, I'm not sure; I think those methods are almost always inlined at runtime but this is a special project...

Reply:
An inner class is regarded as another class in the same package by the JVM before java 11; thus, the main class in theory cannot access private members of another class in the same package. To make the inner class with private field model work, java thus generates a package-private method for *each* usage of these private fields, named like `access$0` or so. Each synthetic method call takes approximately 11 times longer than a direct field access.

Reply:
> Each synthetic method call takes approximately 11 times longer than a direct field access.

No it doesn't. Method calls, stack allocation, etc. is "expensive" compared to direct field accesses, but that's only applicable while the code is ran interpreted, and even then we're talking about _hundreds of nanoseconds_ at most. Don't flatter yourself, your code is not that performance sensitive. And even that's irrelevant, those accessor methods are 100% gone by the time C2 gets to the code, maybe even by C1.

Using package-private for nested fields produces cleaner bytecode, which should be the main reason why you would choose to do that; performance concerns for the accessor methods are irrelevant.